portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/idempotentexecutor/IdempotentImportExecutorExtension.java

```diff
@@ -6,4 +6,5 @@
 
 public interface IdempotentImportExecutorExtension extends BootExtension {
   IdempotentImportExecutor getIdempotentImportExecutor(ExtensionContext extensionContext);
+  IdempotentImportExecutor getRetryingIdempotentImportExecutor(ExtensionContext extensionContext);
```

Comment:
you need the default implementation here (it can either return null or call getIdempotentImportExecutor()..) then you can remove the overrides in the specific implementions

Reply:
Ah ok, that makes sense, I'll change that